ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -711,7 +716,16 @@ private void processGetBlobResponse(GetRequestInfo getRequestInfo, GetResponse g
         } else {
           getError = getResponse.getPartitionResponseInfoList().get(0).getErrorCode();
           if (getError == ServerErrorCode.No_Error) {
-            handleBody(getResponse.getInputStream());
+            PartitionResponseInfo partitionResponseInfo = getResponse.getPartitionResponseInfoList().get(0);
+            int objectsInPartitionResponse = partitionResponseInfo.getMessageInfoList().size();
+            if (objectsInPartitionResponse != 1) {
+              chunkException = new RouterException(
+                  "Unexpected number of messages in a partition response, expected: 1, " + "received: "
+                      + objectsInPartitionResponse, RouterErrorCode.UnexpectedInternalError);
+            }
+
```

Comment:
shouldn't we have an else here. After setting the exception why do we need to proceed ?